solarwinds_apm/oboe/http_sampler.py

```diff
@@ -0,0 +1,109 @@
+# © 2025 SolarWinds Worldwide, LLC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at:http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+from __future__ import annotations
+
+import logging
+import socket
+import threading
+from typing import Any
+
+import requests
+from opentelemetry.sdk.metrics import MeterProvider
+
+from solarwinds_apm.oboe.configuration import Configuration
+from solarwinds_apm.oboe.sampler import Sampler
+
+REQUEST_TIMEOUT = 10  # 10s
+RETRY_INITIAL_TIMEOUT = 0.5  # 500ms
+RETRY_MAX_TIMEOUT = 60  # 60s
+RETRY_MAX_ATTEMPTS = 20
+MULTIPLIER = 1.5
+
+DAEMON_THREAD_JOIN_TIMEOUT = 10  # 10s
+REQUEST_INTERVAL = 60  # 60s
+
+
+class HttpSampler(Sampler):
```

Comment:
Since `HttpSampler` and `JsonSampler` are the 2 concrete classes that APM Python should be using, could we leave their files where they are, then move the rest into a helper subdir for clarity? Not sure what to name the subdir though:

```
solarwinds_apm
├── oboe
│   ├── sampling              <-- new subdir
│   ├── ├── __init__.py       <-- new, additional init
│   ├── ├── configuration.py
│   ├── ├── dice.py
│   ├── ├── metrics.py
│   ├── ├── oboe_sampler.py   <-- concrete, maybe here or in level above?
│   ├── ├── sampler.py        <-- ABC
│   ├── ├── ...
│   ├── __init__.py
│   ├── http_sampler.py
│   ├── json_sampler.py
```

Reply:
Or we can leave flat for now, and do a re-org in a different PR